packages/shared/src/models/examples.ts

```diff
@@ -0,0 +1,59 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+export interface Category {
+  id: string;
+  name: string;
+  description?: string;
+}
+
+// validator for certain types arm / amd64
+export type ArchitectureType = 'arm64' | 'amd64';
+
+export interface Example {
+  id: string;
+  name: string;
+  categories: string[];
+  description: string;
+  repository: string;
+
+  // List of available architectures to pull wtih the image.
+  architectures?: ArchitectureType[];
+
+  // container image to pull, some may not be available... so make this optional.
+  image?: string;
+
+  // tag, else we will assume 'latest'
+  tag?: string;
+
+  // Optional reference version for the example
+  ref?: string;
+
+  // Will assume 'Containerfile' in the root of the repo.
+  basedir?: string;
+
+  // Marks if it is already pulled or not
+  pulled?: boolean;
```

Comment:
These are the only cases of 'state' where everything else comes from the file. Feels a bit odd here since I can't review how the data is merged, wondering if it should be left off until a PR that would set it? Maybe better to have a separate object for state?

Second, pulled/pulling can't both be true so maybe should be an enum? But then maybe we want a 'building disk image' state? Again, might be easier to remove for now and review in context.